soh/soh/Enhancements/randomizer/hook_handlers.cpp

```diff
@@ -619,18 +619,148 @@ void func_8083A434_override(PlayState* play, Player* player) {
+                        itemSkull_loc->SetCheckStatus(RCSHOW_IDENTIFIED);
+                    }
+                    reveal = RC_DEKU_THEATER_MASK_OF_TRUTH;
+                }
+                break;
+            case TEXT_GHOST_SHOP_EXPLAINATION:
+            case TEXT_GHOST_SHOP_CARD_HAS_POINTS:
+                if (ctx->GetOption(RSK_BIG_POES_HINT)) {
+                    reveal = RC_MARKET_10_BIG_POES;
+                }
+                break;
+            case TEXT_MALON_EVERYONE_TURNING_EVIL:
+            case TEXT_MALON_I_SING_THIS_SONG:
+            case TEXT_MALON_HOW_IS_EPONA_DOING:
+            case TEXT_MALON_OBSTICLE_COURSE:
+            case TEXT_MALON_INGO_MUST_HAVE_BEEN_TEMPTED:
+                if (ctx->GetOption(RSK_MALON_HINT)) {
+                    reveal = RC_KF_LINKS_HOUSE_COW;
+                }
+                break;
+            case TEXT_FROGS_UNDERWATER:
+                if (ctx->GetOption(RSK_FROGS_HINT)) {
+                    reveal = RC_ZR_FROGS_OCARINA_GAME;
+                }
+                break;
+            case TEXT_GF_HBA_SIGN:
+            case TEXT_HBA_INITIAL_EXPLAINATION:
+            case TEXT_HBA_ALREADY_HAVE_1000:
+                if (ctx->GetOption(RSK_HBA_HINT)) {
+                    auto item1000_loc = ctx->GetItemLocation(RC_GF_HBA_1000_POINTS);
+                    if (item1000_loc->GetCheckStatus() == RCSHOW_UNCHECKED) {
+                        item1000_loc->SetCheckStatus(RCSHOW_IDENTIFIED);
+                    }
+                    reveal = RC_GF_HBA_1500_POINTS;
+                }
+                break;
+            case TEXT_SCRUB_RANDOM:
+                if (ctx->GetOption(RSK_SCRUB_TEXT_HINT).GetSelectedOptionIndex() != RO_GENERIC_OFF) {
+                    EnDns* enDns = (EnDns*)actor;
+                    reveal = OTRGlobals::Instance->gRandomizer->GetCheckFromRandomizerInf((RandomizerInf)enDns->sohScrubIdentity.randomizerInf);
+                }
+                break;
+            case TEXT_BEAN_SALESMAN_BUY_FOR_10:
+                if (revealMerchant && (ctx->GetOption(RSK_SHUFFLE_MERCHANTS).Is(RO_SHUFFLE_MERCHANTS_BEANS_ONLY) ||
+                  ctx->GetOption(RSK_SHUFFLE_MERCHANTS).Is(RO_SHUFFLE_MERCHANTS_ALL))) {
+                    reveal = RC_ZR_MAGIC_BEAN_SALESMAN;
+                }
+                break;
+            case TEXT_GRANNYS_SHOP:
+                if (!Flags_GetRandomizerInf(RAND_INF_MERCHANTS_GRANNYS_SHOP) && revealMerchant && nonBeanMerchants &&
+                  (ctx->GetOption(RSK_SHUFFLE_ADULT_TRADE) || INV_CONTENT(ITEM_CLAIM_CHECK) == ITEM_CLAIM_CHECK)) {
+                     reveal = RC_KAK_GRANNYS_SHOP;
+                 }
+                break;
+            case TEXT_MEDIGORON:
+                if (revealMerchant && nonBeanMerchants) {
+                    reveal = RC_GC_MEDIGORON;
+                }
+                break;
+            case TEXT_CARPET_SALESMAN_1:
```

Comment:
This is failing to reveal for me, probably because of `!ctx->GetOption(RSK_MERCHANT_TEXT_HINT).GetSelectedOptionIndex() != RO_GENERIC_OFF`. 

You shouldn't need to copy that part of the text logic (which is replicated in revealMerchant, bit not in this broken inverted state) or the rand_inf check (as that's just there to keep default text after the item has been collected.